### PR TITLE
2700 recruitment getallfaqs endpoint

### LINKED ISSUE #2700: [Recruitment] - getAllFAQs endpoint
### Desired Changes

Blocked by #2699 

We want to get all the FAQs so we can display them on our frontend. To do this we need to create the endpoint for it.

### Acceptance Criteria
- Create the endpoint in the /recruitment/faq route 
- Add unit tests to both of them to ensure they work as expected
- Add three pieces of seed data so we can test this out in the future
- Add Screenshots of manual postman tests to ensure they work

### Screenshots (as needed)

_No response_

### PR DESCRIPTION
**Pull request by superhvarn**
## Changes

- Route, controller, and service
- Tests

## Screenshots
Screenshot 2024-08-10 at 4 06 36 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #2700 )

### CODE REVIEW

File: src/backend/src/routes/recruitment.routes.ts [hunk 1]
```diff
@@ -35,4 +35,8 @@ recruitmentRouter.post(
   RecruitmentController.createFaq
 );
 
+recruitmentRouter.get('/faqs', RecruitmentController.getAllFaqs);
+
+recruitmentRouter.get('/milestones', RecruitmentController.getAllMilestones);
```
Comment (Aaryan1203) [inline src/backend/src/routes/recruitment.routes.ts]: Remove this, already there probably from a merge conflict

File: src/backend/src/services/recruitment.services.ts [hunk 2]
```diff
@@ -110,6 +110,23 @@ export default class RecruitmentServices {
...
+   * Gets all FAQs for the given organization Id
+   * @param organizationId organization Id of the faq
+   * @returns all the faqs from the given organization
+   */
+  static async getAllFaqs(organizationId: string) {
+    const allFaqs = await prisma.frequentlyAskedQuestion.findMany({
+      where: { organizationId }
+    });
+
+    if (!organizationId) {
```
Comment (Aaryan1203) [inline src/backend/src/services/recruitment.services.ts]: move this check above the allFaqs. and also don't do !organizationId. First find the organization and check if it exists like its done in other endpoints

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
great job, couple small things and your good

File: src/backend/src/services/recruitment.services.ts [hunk 3]
```diff
@@ -110,6 +110,27 @@ export default class RecruitmentServices {
...
   }
 
+  /**
+   * Gets all FAQs for the given organization Id
+   * @param organizationId organization Id of the faq
+   * @returns all the faqs from the given organization
+   */
+  static async getAllFaqs(organizationId: string) {
+    const organization = await prisma.organization.findUnique({
+      where: { organizationId }
```
Comment (Aaryan1203) [inline src/backend/src/services/recruitment.services.ts:120]: can you also add where date deleted is null so we don't get any deleted faqs

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
one more small thing and should be good

**Review by Aaryan1203 — DISMISSED** [review]
lgtm

**Review by walker-sean — DISMISSED** [review]
LGTM

**Review by walker-sean — COMMENTED** [review]
Merge conflicts then good 2 go

**Pull request merged**